InternalDocs/stack_protection.md

```diff
@@ -0,0 +1,61 @@
+# Stack Protection
+
+CPython protects against stack overflow in the form of runaway, or just very deep, recursion by raising a `RecursionError` instead of just crashing.
+Protection against pure Python stack recursion has existed since very early, but in 3.12 we added protection against stack overflow
+in C code. This was initially implemented using a counter and later improved in 3.14 to use the actual stack depth.
+For those platforms that support it (Windows, Mac, and most Linuxes) we query the operating system to find the stack bounds.
+For other platforms we use conserative estimates.
+
+
+The C stack looks like this:
+
+```
+         +-------+  <--- Top of machine stack
+         |       |
+         |       |
+
+            ~~
+
+         |       |
+         |       |
+         +-------+  <--- Soft limit
+         |       |
+         |       |     _PyOS_STACK_MARGIN_BYTES
+         |       |
+         +-------+  <---  Hard limit
+         |       |
+         |       |     _PyOS_STACK_MARGIN_BYTES
+         |       |
+         +-------+  <--- Bottom of machine stack
```

Comment:
Remove the indentation?
```suggestion
+-------+  <--- Top of machine stack
|       |
|       |

  ~~~~

|       |
|       |
+-------+  <--- Soft limit
|       |
|       |     _PyOS_STACK_MARGIN_BYTES
|       |
+-------+  <---  Hard limit
|       |
|       |     _PyOS_STACK_MARGIN_BYTES
|       |
+-------+  <--- Bottom of machine stack
```

Reply:
I think it looks better with a bit of whitespace to the left.
https://github.com/python/cpython/blob/a4ffb9ec8ce324689aeb38742baa9d10416ffab2/InternalDocs/stack_protection.md